### ISSUE TITLE
Word Count Tracker

### ISSUE THREAD

**Issue by cengels** [cengels/skywriter#9]
## Description

An indicator that tracks the words written per day.

## Thoughts

- The counter should not immediately reset upon hitting midnight, as many writers work through midnight into the early hours of the morning. Instead, the user may set a fixed time when to reset the counter, or the application could reset it automatically when it is restarted after midnight.
- Optionally, the progress per day may also be stored to create a history.

## To-Do

- [x] Building a CSV parser that can serialize an arbitrary object to and from CSV
- [x] Building basic progress tracking functionality (defining progress items etc.)
- [x] Discovering word boundaries and counting the words based on that
- [x] Building a collapsible status bar that shows the current progress while writing
- [x] A way to undo a deletion (by manually adjusting it)

**Comment by cengels** [comment]
The previous implementation of `ProgressItem` included two fields: `wordsAdded` and `wordsDeleted` where only `wordsAdded` would be counted as part of the statistics, and `wordsDeleted` would serve as a kind of "corrective" value.

Firstly, it has been decided to discard the `wordsDeleted` field entirely and rename `wordsAdded` to simply `words`, the reason being is that, since `wordsDeleted` specifically isn't included in any statistics, storing this value locally makes no sense either.

Secondly, and far more importantly, `wordsDeleted` was previously a calculated value. For instance, if the user deleted words while they were at a `wordsAdded` value of 0, `wordsDeleted` would automatically be adjusted instead of `wordsAdded`.

There are several problems with this approach. One thing is that it is very difficult to nigh-on impossible to track the actual deleted text versus new text. This is necessary to make sure that, for instance, deleting text at a word count of 0 and then undoing that deletion does not increment `wordsAdded` (which, in the previous implementation, was a known bug).

The same problem would arise in many other places as well. If the algorithm were to automatically add any pasted words as corrected words rather than `wordsAdded`, how would it be able to distinguish between text that was cut and pasted from within the document itself? If the original state of the document is not tracked, then cutting words and pasting them over and over again would continuously increment the word count.

Tracking the original state of the document is not as simple as that either, though. How would the algorithm detect if a number of words were cut and pasted from within the document? Common words and phrases like `the` or `he said` are not exactly unique in a document of any size, so a simple `contains()` check would return the wrong result. Checking for position is similarly impossible, though—what if the user inserts more paragraphs and text before the cut words?

In summary, it was decided to leave the tracking of "corrected" word counts in the user's hands via the functions *Delete Untracked* and *Paste Untracked*. This is, as far as I can see, the only way to guarantee that no strange anomalies show up in the user's word count statistics.
